File: a02_whirlpool/java/com/whirlpool/cooking/ViewHolderHelpers/VerticalTumblerViewHolderHelper.java
package com.whirlpool.cooking.ViewHolderHelpers;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.whirlpool.cooking.R;
import com.whirlpool.cooking.base.AbstractViewHolders.AbstractVerticalTumblerViewHolder;
import com.whirlpool.cooking.databinding.FragmentVerticalTumblerBinding;
import com.whirlpool.cooking.widgets.header.HeaderBar;
import com.whirlpool.cooking.widgets.header.HeaderSingleIconDoubleText;
import com.whirlpool.hmi.uicomponents.widgets.tumblers.BaseTumbler;

public class VerticalTumblerViewHolderHelper extends AbstractVerticalTumblerViewHolder {

    private FragmentVerticalTumblerBinding fragmentVerticalTumblerBinding;

    // ================================================================================================================
    // -----------------------------------------------  General Methods  ----------------------------------------------
    // ================================================================================================================

    /**
     * onCreateView provides the interface for the Fragment to be able to inflate the customized view.
     *
     * @param inflater           {@link LayoutInflater}
     * @param container          {@link ViewGroup}
     * @param savedInstanceState {@link Bundle}
     * @return {@link View}
     */
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {

        fragmentVerticalTumblerBinding = FragmentVerticalTumblerBinding.inflate(inflater);

        return fragmentVerticalTumblerBinding.getRoot();
    }

    /*---------------------------------------------------------------------------------------------------------------*/

    /**
     * Provides the ability to clean up the view holder when it is destroyed
     */
    @Override
    public void onDestroyView() {
        fragmentVerticalTumblerBinding = null;
    }

    /*---------------------------------------------------------------------------------------------------------------*/

    /**
     * Provides the interface to access the binding class
     *
     * @return {@link FragmentVerticalTumblerBinding}
     */
    @Override
    public FragmentVerticalTumblerBinding getFragmentVerticalTumblerBinding() {
        return fragmentVerticalTumblerBinding;
    }

    /*---------------------------------------------------X---X---X---------------------------------------------------*/

    // ================================================================================================================
    // ------------------------------------------------  Common Views  ------------------------------------------------
    // ================================================================================================================

    /**
     * Provides the interface to access HeaderBar widget
     *
     * @return {@link HeaderBar}
     */
    @Override
    public HeaderSingleIconDoubleText provideVerticalTumblerHeaderBar() {
        return fragmentVerticalTumblerBinding.widgetHeaderTopmenu;
    }

    /*---------------------------------------------------------------------------------------------------------------*/

    /**
     * Provides the interface to access HoursBaseTumbler widget
     *
     * @return {@link BaseTumbler}
     */
    @Override
    public BaseTumbler provideVerticalHoursTumbler() {
        return fragmentVerticalTumblerBinding.tumblerNumericBasedHours;
    }

    /*---------------------------------------------------------------------------------------------------------------*/

    /**
     * Provides the interface to access MinutesBaseTumbler widget
     *
     * @return {@link BaseTumbler}
     */
    @Override
    public BaseTumbler provideVerticalMinutesTumbler() {
        return fragmentVerticalTumblerBinding.tumblerNumericBasedMins;
    }

    /*---------------------------------------------------------------------------------------------------------------*/

    /**
     * Provides the interface to access SaveTextView
     *
     * @return {@link TextView}
     */
    @Override
    public TextView provideVerticalTumblerSaveTextView() {
        return fragmentVerticalTumblerBinding.cooktimeButtonBottomLeft;
    }

    /*---------------------------------------------------------------------------------------------------------------*/

    /**
     * Provides the interface to access StartTextView
     *
     * @return {@link TextView}
     */
    @Override
    public TextView provideVerticalTumblerStartTextView() {
        return fragmentVerticalTumblerBinding.cooktimeButtonBottomRight;
    }

    public TextView provideLeftUnitTextView() {
        return fragmentVerticalTumblerBinding.txtUnitLeft;
    }

    public TextView provideRightUnitTextView() {
        return fragmentVerticalTumblerBinding.txtUnitRight;
    }

    /*---------------------------------------------------X---X---X---------------------------------------------------*/

    // ================================================================================================================
    // -------------------------------------------  Modality Specific Views  ------------------------------------------

    /**
     * @return {@link int}
     */
    @Override
    public int provideLeftIcon() {
        return R.drawable.icon_40px_backarrow;
    }
    // ================================================================================================================

    /*---------------------------------------------------X---X---X---------------------------------------------------*/
}

File: CookingApplianceM63KA/app/src/main/java/com/whirlpool/cooking/ka/ktn/core/viewHolderHelpers/VerticalTumblerViewHolderHelper.kt
/*
 *
 * * ************************************************************************************************
 * * ***** Copyright (c) 2024. Whirlpool Corporation. All rights reserved - CONFIDENTIAL *****
 * * ************************************************************************************************
 *
 */
package core.viewHolderHelpers

import android.os.Bundle
import android.presenter.customviews.textButton.TextButton
import android.presenter.customviews.widgets.headerbar.HeaderBarWidget
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.constraintlayout.widget.ConstraintLayout
import com.whirlpool.cooking.ka.databinding.VerticalFragmentTumblerBinding
import com.whirlpool.hmi.uicomponents.widgets.resourcehelper.ResourceTextView
import com.whirlpool.hmi.uicomponents.widgets.tumblers.BaseTumbler
import core.jbase.abstractViewHolders.AbstractVerticalTumblerViewHolder

/**
 * File        : com.whirlpool.cooking.ka.ktn.core.viewHolderHelpers
 * Brief       : This class will provide base basic vertical tumbler implementation and by extending this class we can make changes into UI and functionality further.
 * Author      : SINGHJ25
 * Created On  : 18/02/2024
 */
class VerticalTumblerViewHolderHelper :
    AbstractVerticalTumblerViewHolder() {
    private var fragmentTumblerBinding: VerticalFragmentTumblerBinding? = null

    override fun onCreateView(
        inflater: LayoutInflater?,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        fragmentTumblerBinding = VerticalFragmentTumblerBinding.inflate(inflater!!)
        return fragmentVerticalTumblerBinding?.root
    }

    /**
     * Provides the binding class
     *
     * @return [VerticalFragmentTumblerBinding]
     */
    override val fragmentVerticalTumblerBinding: VerticalFragmentTumblerBinding?
        get() = fragmentTumblerBinding

    override fun onDestroyView() {
        fragmentTumblerBinding = null
    }

    override fun provideGhostButton(): TextButton? {
        return fragmentTumblerBinding?.btnGhost
    }

    override fun provideConstraintGhost(): ConstraintLayout? {
        return fragmentTumblerBinding?.constraintGhostLeft
    }

    override fun provideLeftPowerButton(): TextButton? {
        return fragmentTumblerBinding?.verticalTumblerCookTimeLeftPower
    }

    override fun provideLeftPowerConstraint(): ConstraintLayout? {
        return fragmentTumblerBinding?.constraintGhostLeftPower
    }

    override fun providePrimaryButton(): TextButton? {
        return fragmentTumblerBinding?.btnPrimary
    }

    override fun providePrimaryConstraint(): ConstraintLayout? {
        return fragmentTumblerBinding?.constraintPrimaryRight
    }

    override fun provideVerticalTumblerLeft(): BaseTumbler? {
        return fragmentTumblerBinding?.tumblerLeft
    }

    override fun provideVerticalTumblerCenter(): BaseTumbler? {
        return fragmentTumblerBinding?.tumblerCenter
    }

    override fun provideLeftTumblerText(): ResourceTextView? {
        return fragmentTumblerBinding?.labelLeftTumbler
    }

    override fun provideCenterTumblerText(): ResourceTextView? {
        return fragmentTumblerBinding?.labelCenterTumbler
    }

    override fun provideTextViewHelperText(): TextView? {
        return fragmentTumblerBinding?.textViewHelperText
    }

    override fun provideHeaderBar(): HeaderBarWidget? {
        return fragmentTumblerBinding?.headerBar
    }
}